packages/aws_common/lib/src/io/aws_file_platform_io.dart

```diff
@@ -104,4 +108,34 @@ class AWSFilePlatform extends AWSFile {
     // the constructors ensures it won't reach to this point, but just in case
     throw const InvalidFileException();
   }
+
+  @override
+  Future<String?> get contentType async {
+    if (_hasResolvedContentType) {
+      return _resolvedContentType;
+    }
```

Comment:
You don't need the extra variable

```suggestion
    final resolvedContentType = _resolvedContentType;
    if (resolvedContentType != null) {
      return resolvedContentType;
    }
```

Reply:
Same for web

Reply:
If you use `AsyncMemoizer` or `late final` you don't need `_resolvedContentType` either